### PR TITLE
add support for update and delete schema field

### PR DESCRIPTION
**Pull request by jkaho**

### CODE REVIEW

**Comment by jkaho** [comment]
@jonathaningram this includes a bunch of stuff I'm not sure if we want to include?

**Review by jonathaningram — CHANGES_REQUESTED** [review]
Let's tidy the openapi.json file up first.

File: docs/CollectionsApi.md [hunk 1]
```diff
@@ -17,7 +17,7 @@ All URIs are relative to *https://api.search.io*
 
 ## create_collection
 
-> <Collection> create_collection(collection_id, collection)
+> <Collection> create_collection(collection_id, collection, opts)
```
Comment (jkaho) [inline docs/CollectionsApi.md:20]: @jonathaningram this an incompatible API change?

Comment (jonathaningram) [reply docs/CollectionsApi.md:20]: @jkaho I don't think so. See https://github.com/sajari/sdk_ruby/pull/4/files?show-viewed-files=true&file-filters%5B%5D=#diff-15a79a26c88566b49854a47c39549808fddec67a725a79fb32866dc5a12a70bfR29

The `opts` have a default value. Would still be a minor version bump though.

**Review by jonathaningram — CHANGES_REQUESTED** [review]
RCFV

File: docs/CollectionsApi.md (hunk 1 above)
Comment (jkaho) [reply docs/CollectionsApi.md:20]: @jonathaningram ah okay. Which file/line does that link to? I'm getting `There are no files selected for viewing`

Comment (jonathaningram) [reply docs/CollectionsApi.md:20]: @jkaho lib/sdk_ruby/api/collections_api.rb:29

**Review by jonathaningram — CHANGES_REQUESTED** [review]
RCFV

**Review by jonathaningram — APPROVED** [review]
Build fails but seems to for previous PRs too so doesn't need to block this change.

**Pull request merged**